### PR TITLE
fix: swap hover states

### PR DESCRIPTION
**Pull request by vm**
Screen Shot 2022-10-04 at 12 29 33 PM
Screen Shot 2022-10-04 at 12 29 41 PM
Screen Shot 2022-10-04 at 12 29 46 PM

### CODE REVIEW

File: src/components/swap/SwapDetailsDropdown.tsx [hunk 1]
```diff
@@ -23,9 +23,12 @@ import { ResponsiveTooltipContainer } from './styleds'
 import SwapRoute from './SwapRoute'
 import TradePrice from './TradePrice'
 
-const Wrapper = styled(Row)`
+const Wrapper = styled(Row)<{ redesignFlag: boolean }>`
   width: 100%;
   justify-content: center;
+  background-color: ${({ theme, redesignFlag }) => (redesignFlag ? theme.backgroundFloating : 'none')};
+  padding: ${({ redesignFlag }) => (redesignFlag ? '8px 12px' : 'none')};
+  margin-top: 4px !important;
```
Comment (vm) [inline src/components/swap/SwapDetailsDropdown.tsx:31]: rm

File: src/components/swap/SwapDetailsDropdown.tsx [hunk 2]
```diff
@@ -41,14 +45,11 @@ const StyledCard = styled(OutlineCard)<{ redesignFlag: boolean }>`
 `
 
 const StyledHeaderRow = styled(RowBetween)<{ disabled: boolean; open: boolean; redesignFlag: boolean }>`
-  padding: ${({ redesignFlag }) => (redesignFlag ? '8px 0px 0px 0px' : '4px 8px')};
+  padding: ${({ redesignFlag }) => (redesignFlag ? 'none' : '4px 8px')};
```
Comment (zzmp) [inline src/components/swap/SwapDetailsDropdown.tsx:48]: nit: I prefer `0` over `none`.

File: src/components/swap/SwapDetailsDropdown.tsx [hunk 3]
```diff
@@ -133,7 +134,7 @@ export default function SwapDetailsDropdown({
   const redesignFlagEnabled = redesignFlag === RedesignVariant.Enabled
 
   return (
-    <Wrapper style={{ marginTop: '8px' }}>
+    <Wrapper style={{ marginTop: '0px' }} redesignFlag={redesignFlagEnabled}>
```
Comment (zzmp) [inline src/components/swap/SwapDetailsDropdown.tsx:137]: Does this need to retain a `marginTop` without `redesignFlagEnabled`?

File: src/pages/Swap/index.tsx [hunk 4]
```diff
@@ -463,7 +497,7 @@ export default function Swap() {
   const swapIsUnsupported = useIsSwapUnsupported(currencies[Field.INPUT], currencies[Field.OUTPUT])
 
   const priceImpactTooHigh = priceImpactSeverity > 3 && !isExpertMode
-  const showPriceImpactWarning = largerPriceImpact && priceImpactSeverity > 3
+  const showPriceImpactWarning = largerPriceImpact && priceImpactTooHigh
```
Comment (zzmp) [inline src/pages/Swap/index.tsx:500]: This will no longer show the impact warning in expert mode. Is this intended?

**Review by zzmp — APPROVED** [review]
lgtm, a few minor questions

**Review by zzmp — COMMENTED** [review]
There's something slowing down rendering on hover - I'm looking into the preview rn

**Review by zzmp — CHANGES_REQUESTED** [review]
Hovering over the swap-arrows button makes it transparent, and shows the outline of the button below it. There should be an opaque box underneath the button to prevent this leak-through.

**Comment by vm** [comment]
1) The hover issue is present on prod as well. I reverted the arrow changes (other than an alignment fix) and will copy what is in widget over to fix.
2) Checked the design and the reverse button goes over the border.

**Review by zzmp — APPROVED** [review]

**Pull request merged**